lisa/tools/hwclock.py

```diff
@@ -25,3 +29,20 @@ def install(self) -> bool:
     def set_rtc_clock_to_system_time(self) -> None:
         cmd_result = self.run("--systohc", shell=True, sudo=True)
         cmd_result.assert_exit_code()
+
+    def set_datetime(self, time: datetime) -> None:
+        cmd_result = self.run(
+            f"--set --date='{time}'", shell=True, sudo=True, force_run=True
+        )
+        cmd_result.assert_exit_code()
+
+    @retry(exceptions=LisaException, tries=20, delay=0.5)  # type: ignore
+    def get(self, no_error_log: bool = True) -> datetime:
+        command_result = self.run(
```

Comment:
We can use the exit code check parameter in `run` function

Reply:
fixed